openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java

```diff
@@ -0,0 +1,80 @@
+package org.openmetadata.service.util;
+
+import io.dropwizard.db.DataSourceFactory;
+import io.dropwizard.jdbi3.JdbiFactory;
+import io.dropwizard.setup.Environment;
+import java.net.URI;
+import java.time.temporal.ChronoUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.SqlLogger;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.sqlobject.SqlObjects;
+import org.openmetadata.service.OpenMetadataApplicationConfig;
+import org.openmetadata.service.jdbi3.locator.ConnectionAwareAnnotationSqlLocator;
+
+@Slf4j
+public final class JdbiUtils {
+
+  private JdbiUtils() {}
+
+  public static Jdbi createJdbi(OpenMetadataApplicationConfig config, Environment environment) {
+
+    // Prepare
+    DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
+
+    // Check auth provider
+    checkAuthenticationProvider(config, dataSourceFactory);
+
+    // Build
+    Jdbi jdbi = new JdbiFactory().build(environment, dataSourceFactory, "database");
+
+    // Set logging
+    setLogging(dataSourceFactory, jdbi);
+
+    // Return
+    return jdbi;
+  }
+
+  private static void checkAuthenticationProvider(
+      OpenMetadataApplicationConfig config, DataSourceFactory dataSourceFactory) {
+    // Check auth provider
+    if (config.getAwsConfiguration() != null && config.getAwsConfiguration().getEnableIamDatabaseAuthentication()) {
+      // Parse jdbc url
+      URI uri = URI.create(dataSourceFactory.getUrl().substring(5));
```

Comment:
what's the substring doing here and why there is 5 chars. Can we add some comment on this number